wiring/src/spark_wiring_ble.cpp

```diff
@@ -308,35 +316,49 @@ BleUuid::BleUuid(const String& uuid)
 
 bool BleUuid::isValid() const {
     if (type() == BleUuidType::SHORT) {
-        return uuid_.uuid16 != 0x0000;
+        return (uuid128_[UUID16_LO] != 0x00 || uuid128_[UUID16_HI] != 0x00);
     } else {
-        uint8_t temp[BLE_SIG_UUID_128BIT_LEN] = {};
-        return memcmp(uuid_.uuid128, temp, BLE_SIG_UUID_128BIT_LEN);
+        return memcmp(uuid128_, BASE_UUID, BLE_SIG_UUID_128BIT_LEN);
     }
 }
 
 BleUuidType BleUuid::type() const {
-    if (uuid_.type == BLE_UUID_TYPE_16BIT || uuid_.type == BLE_UUID_TYPE_128BIT_SHORTED) {
-        return BleUuidType::SHORT;
-    } else {
-        return BleUuidType::LONG;
-    }
+    return type_;
 }
 
 hal_ble_uuid_t BleUuid::halUUID() {
-    return uuid_;
+    hal_ble_uuid_t uuid = {};
+    if (type_ == BleUuidType::SHORT) {
+        uuid.type = BLE_UUID_TYPE_16BIT;
+        uuid.uuid16 = (uint16_t)uuid128_[UUID16_LO] | ((uint16_t)uuid128_[UUID16_HI] << 8);
+    } else {
+        uuid.type = BLE_UUID_TYPE_128BIT;
+        memcpy(uuid.uuid128, uuid128_, BLE_SIG_UUID_128BIT_LEN);
+    }
+    return uuid;
 }
 
 uint16_t BleUuid::shorted() const {
-    return uuid_.uuid16;
+    return ((uint16_t)uuid128_[UUID16_LO] | ((uint16_t)uuid128_[UUID16_HI] << 8));
 }
 
-void BleUuid::rawBytes(uint8_t uuid128[BLE_SIG_UUID_128BIT_LEN]) const {
-    memcpy(uuid128, uuid_.uuid128, BLE_SIG_UUID_128BIT_LEN);
+size_t BleUuid::rawBytes(uint8_t uuid128[BLE_SIG_UUID_128BIT_LEN]) const {
+    if (type() == BleUuidType::SHORT) {
```

Comment:
I thought we agreed that this should always return full 128-bit UUID and apply the base uuid to the shortened ones?